### PR TITLE
Revert the previous pre booking notification changes and added new code changes on Mec dev 1.2.0.1 b1

### PR DESCRIPTION
**Pull request by AvanishDuggiredddy**

### CODE REVIEW

File: pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/service/DemographicService.java [hunk 1]
```diff
@@ -250,8 +254,13 @@ public class DemographicService implements DemographicServiceIntf {
...
 	@Value("${mosip.utc-datetime-pattern}")
 	private String mosipDateTimeFormat;
-
-
+	
+	@Autowired
+	NotificationService notificationService;
+	
+	@Autowired
+	private ObjectMapper objectMapper;
```
Comment (MuralitharanK) [inline pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/service/DemographicService.java:259]: Object mapper in 2 places, remove one place

File: pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/service/NotificationService.java [hunk 2]
```diff
@@ -25,6 +19,12 @@
 import org.springframework.web.client.RestClientException;
 import org.springframework.web.multipart.MultipartFile;
 
+import com.fasterxml.jackson.databind.JsonNode;
```
Comment (MuralitharanK) [inline pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/service/NotificationService.java:22]: remove unused imports

File: pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/service/NotificationService.java [hunk 3]
```diff
@@ -375,47 +392,57 @@ public MainResponseDTO<DemographicResponseDTO> notificationDtoValidation(Notific
...
-		if (responseEntity.getErrors() != null) {
-			throw new DemographicDetailsNotFoundException(responseEntity.getErrors(), response);
-		}
-		JsonNode responseNode = objectMapper.readTree(responseEntity.getResponse().getDemographicDetails().toJSONString());
-		responseNode = responseNode.get(identity);
-		if (!notificationDto.isAdditionalRecipient()) {
-			if (notificationDto.getMobNum() != null || notificationDto.getEmailID() != null) {
-				log.error("sessionId", "idType", "id",
-						"Not considering the requested mobilenumber/email since additional recipient is false ");
+		if (!(prid != null)) {
```
Comment (MuralitharanK) [inline pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/service/NotificationService.java]: this is not correct

File: pre-registration/pre-registration-application-service/src/main/resources/application.properties [hunk 4]
```diff
@@ -64,9 +64,6 @@ document.service.version: v1
 email.acknowledgement.subject.template: Acknowledgement-email-subject
 email.acknowledgement.template: Email-Acknowledgement
 cancel.appointment.email.subject: cancel-appointment-email-subject
-email.prebooking.acknowledgement.template: Pre-Booking-PreReg-Email-Notification
```
Comment (MuralitharanK) [inline pre-registration/pre-registration-application-service/src/main/resources/application.properties]: why you removed these templates?

Comment (AvanishDuggiredddy) [reply pre-registration/pre-registration-application-service/src/main/resources/application.properties]: will update all the changes

**Review by MuralitharanK — APPROVED** [review]

**Pull request merged**